diag_manager/fms_diag_object.F90

```diff
@@ -425,6 +429,58 @@ FUNCTION fms_diag_axis_init(this, axis_name, axis_data, units, cart_name, long_n
 #endif
 end function fms_diag_axis_init
 
+!> Accepts data from the send_data functions.  If this is in an openmp region with more than
+!! one thread, the data is buffered in the field object and processed later.  If only a single thread
+!! is being used, then the processing can be done and stored in the buffer object.  The hope is that
+!! the increase in memory footprint related to buffering can be handled by the shared memory of the
+!! multithreaded case.
+!! \note If some of the diag manager is offloaded in the future, then it should be treated similarly
+!! to the multi-threaded option for processing later
+logical function fms_diag_accept_data (this, diag_field_id, field, time, is_in, js_in, ks_in, &
+                  mask, rmask, ie_in, je_in, ke_in, weight, err_msg)
+  class(fmsDiagObject_type),TARGET,INTENT(inout):: this !< Diaj_obj to fill
+  INTEGER, INTENT(in) :: diag_field_id !< The ID of the input diagnostic field
+  CLASS(*), DIMENSION(:,:,:,:), INTENT(in) :: field !< The data for the input diagnostic
```

Comment:
I think a better name for "field" here  is "field_data" as the word field is practically synonymous with fmsDiagField_type.